src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -1,2760 +1,1932 @@
+            dstarlya_cont_dt_prefactor[R_ct] = zpp_integrand * sum_ly2_val;
+            dstarlya_inj_dt_prefactor[R_ct] = zpp_integrand * sum_lynto2_val;
+            // LOG_DEBUG("cont %.2e inj %.2e",dstarlya_cont_dt_prefactor[R_ct],dstarlya_inj_dt_prefactor[R_ct]);
+        }
+        if(flag_options_ts->USE_MINI_HALOS){
+            dstarlya_dt_prefactor_MINI[R_ct]  = zpp_integrand * sum_lyn_val_MINI;
+            dstarlyLW_dt_prefactor[R_ct] = zpp_integrand * sum_lyLW_val;
+            dstarlyLW_dt_prefactor_MINI[R_ct]  = zpp_integrand * sum_lyLW_val_MINI;
+            if(flag_options_ts->USE_LYA_HEATING){
+                dstarlya_cont_dt_prefactor_MINI[R_ct] = zpp_integrand * sum_ly2_val_MINI;
+                dstarlya_inj_dt_prefactor_MINI[R_ct] = zpp_integrand * sum_lynto2_val_MINI;
+            }
+
+            // LOG_DEBUG("starmini: %.2e LW: %.2e LWmini: %.2e",dstarlya_dt_prefactor_MINI[R_ct],
+            //                                             dstarlyLW_dt_prefactor[R_ct],
+            //                                             dstarlyLW_dt_prefactor_MINI[R_ct]);
+            // LOG_DEBUG("cont mini %.2e inj mini %.2e",dstarlya_cont_dt_prefactor_MINI[R_ct],dstarlya_inj_dt_prefactor_MINI[R_ct]);
         }
-    }
-
-    init_ps();
 
-LOG_SUPER_DEBUG("Initialised PS");
-LOG_SUPER_DEBUG("About to initialise heat");
-    init_heat();
-LOG_SUPER_DEBUG("Initialised heat");
+        sum_lyn_prev = sum_lyn_val;
+        sum_lyn_prev_MINI = sum_lyn_val_MINI;
+        sum_ly2_prev = sum_ly2_val;
+        sum_ly2_prev_MINI = sum_ly2_val_MINI;
+        sum_lynto2_prev = sum_lynto2_val;
+        sum_lynto2_prev_MINI = sum_lynto2_val_MINI;
+    }
+    LOG_DEBUG("R0 starlya: %.4e",dstarlya_dt_prefactor[0]);
+}
 
-    // Initialize some interpolation tables
-    // if(initialization_required) {
+//fill fftwf boxes, do the r2c transform and normalise
+void prepare_filter_boxes(double redshift, float *input_dens, float *input_vcb, float *input_j21, fftwf_complex *output_dens, fftwf_complex *output_LW){
+    int i,j,k,ct;
+    double curr_vcb,curr_j21,M_buf;
 
-        if(user_params->USE_INTERPOLATION_TABLES) {
-          if(user_params->FAST_FCOLL_TABLES){
-            initialiseSigmaMInterpTable(fmin(MMIN_FAST,M_MIN),1e20);
-          }
-          else{
-            if(flag_options->M_MIN_in_Mass || flag_options->USE_MASS_DEPENDENT_ZETA) {
-                if (flag_options->USE_MINI_HALOS){
-                    initialiseSigmaMInterpTable(global_params.M_MIN_INTEGRAL/50.,1e20);
-                }
-                else{
-                    initialiseSigmaMInterpTable(M_MIN,1e20);
-                }
+    //TODO: Meraxes just applies a pointer cast box = (fftwf_complex *) input. Figure out why this works,
+    //      They pad the input by a factor of 2 to cover the complex part, but from the type I thought it would be stored [(r,c),(r,c)...]
+    //      Not [(r,r,r,r....),(c,c,c....)] so the alignment should be wrong, right?
```

Comment:
Not yet, but I'm 99% sure its just my lack of understanding, meraxes just does a cast (albeit with a `//CAREFUL!!!!!` comment). It would be nice to save the memory and time on the reinitialisaiton. I plan on making some small tests to see how/when this works